### PR TITLE
@dzucconi: add new properties on artist schema

### PR DESCRIPTION
**Pull request by xtina-starr**
This adds a few new properties on the Artist object which are needed per [#1260](https://github.com/artsy/microgravity/issues/1260) this spec. The overview section is a bit premature, but later when the show logic for the artist is complete I think the overview property will be needed.

### CODE REVIEW

File: schema/artist/index.js [hunk 1]
```diff
@@ -88,6 +94,38 @@ const ArtistType = new GraphQLObjectType({
               type: GraphQLInt,
               resolve: ({ auction_lots_count }) => auction_lots_count,
             },
+            forsale_artworks_count: {
```
Comment (dzucconi) [inline schema/artist/index.js]: I would rename this as `for_sale_artworks`. the `count` is implicit in the parent field

File: schema/artist/index.js [hunk 2]
```diff
@@ -88,6 +94,38 @@ const ArtistType = new GraphQLObjectType({
...
+                  return `${artist.years}, lives and works in ${artist.location}`
+                } else if (artist.years) {
+                  return `${artist.years}`
+                } else if (artist.hometown && artist.location) {
+                  return `${artist.hometown}, lives and works in ${artist.location}`
+                } else if (artist.hometown) {
+                  return `${artist.hometown}`
+                } else if (artist.location) {
+                  return `${artist.location}`
+                }
```
Comment (dzucconi) [inline schema/artist/index.js]: Much more simply expressed as:

```javascript
    return compact([
      years,
      hometown,
      (location ? `lives and works in ${location}` : undefined),
    ]).join(', ');
```

File: schema/artist/index.js [hunk 3]
```diff
@@ -88,6 +94,38 @@ const ArtistType = new GraphQLObjectType({
...
+              type: GraphQLInt,
+              resolve: ({ forsale_artworks_count }) => forsale_artworks_count,
+            },
+          },
+        }),
+        resolve: (artist) => artist,
+      },
+      overview: {
+        type: new GraphQLObjectType({
+          name: 'ArtistOverview',
```
Comment (dzucconi) [inline schema/artist/index.js]: I'm not so sure this has to exist as a sub-object like this. The bio seems fine on the artist schema itself.

**Comment by dzucconi** [comment]
I would also install sublimelinter & eslint—which will help catch the things that are failing inline:

File: schema/artist/index.js [hunk 4]
```diff
@@ -88,6 +94,38 @@ const ArtistType = new GraphQLObjectType({
...
+          fields: {
+            bio: {
+              type: GraphQLString,
+              resolve: (artist) => {
+                if (artist.years && artist.hometown && artist.location) {
```
Comment (xtina-starr) [inline schema/artist/index.js]: Thinking there's gotta be a better way to do this loaded conditional. I thought about creating an array and removing null, undefined, and ' ', value then joining but that didn't really feel right either.

File: schema/artist/index.js (hunk 2 above)
Comment (xtina-starr) [inline schema/artist/index.js]: Ahhh, yes!!!

File: schema/artist/index.js (hunk 4 above)
Comment (dzucconi) [inline schema/artist/index.js]: See below, that's a fine idea and much cleaner

File: schema/artist/index.js (hunk 2 above)
Comment (xtina-starr) [inline schema/artist/index.js]: I like this solution a lot. It kind of what I had in my mind but couldn't quite get out. Thanks for the suggestions!

**Comment by dzucconi** [comment]
Looks good! Thanks for the edits

**Pull request merged**